### PR TITLE
Enable GKE node-pool autoscaling options

### PR DESCRIPTION
**Pull request by richardsliu**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1273)

### CODE REVIEW

File: scripts/gke/deploy.sh [hunk 1]
```diff
@@ -159,6 +166,12 @@ if ${KUBEFLOW_DEPLOY}; then
 
   # Install the GPU driver. It has no effect on non-GPU nodes.
   kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/stable/nvidia-driver-installer/cos/daemonset-preloaded.yaml
+
+  # Update the cluster if autoscaling is desired.
+  if ${ENABLE_AUTOSCALE}; then
+    gcloud container clusters update ${DEPLOYMENT_NAME} --enable-autoscaling --min-nodes ${DEFAULT_MIN_NODES} --max-nodes ${DEFAULT_MAX_NODES} --zone ${ZONE} --node-pool default-pool
```
Comment (jlewi) [inline scripts/gke/deploy.sh]: We want all configuration to be declarative. Can we add this to the CPU & GPU node pools respectively
https://github.com/kubeflow/kubeflow/blob/master/scripts/gke/deployment_manager_configs/cluster.jinja#L116

**Comment by richardsliu** [comment]
Done.

**Comment by jlewi** [comment]
@richardsliu I think you want to rebase off the latest master to fix the tests.

**Comment by richardsliu** [comment]
fixes #1033

**Pull request merged**